src/wechaty_plugin_contrib/contrib/chat_history_plugin/plugin.py

```diff
@@ -36,46 +60,38 @@ def __init__(self, options: Optional[ChatHistoryPluginOptions] = None):
         if not os.path.exists(self.chat_history_path):
             os.makedirs(self.chat_history_path)
-        self.database_path = os.path.join(
-            self.chat_history_path, 'chathistory.db')
+        if not options.chat_history_database:
+            self.chat_history_database = 'sqlite+aiosqlite:///chathistory.db'
+        else:
+            self.chat_history_database = options.chat_history_database
 
     @property
     def name(self) -> str:
         return 'chat-history'
 
     async def on_message(self, msg: Message):
         """listen message event"""
-        msg_id = msg.message_id
-        msg_date = msg.date()
-        msg_talker = str(msg.talker()) if msg.payload.from_id else None
-        msg_to = str(msg.to()) if msg.payload.to_id else None
-        msg_room = str(msg.room()) if msg.payload.room_id else None
-        msg_text = msg.text()
-        msg_type = msg.type()
-        message = (msg_date, msg_id, msg_type,
-                   msg_talker, msg_to, msg_room, msg_text)
-
-        conn = await aiosqlite.connect(self.database_path)
-        create_database_sql = """
-        CREATE TABLE IF NOT EXISTS ChatHistory(
-            id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
-            msg_date DATETIME DEFAULT NULL,
-            msg_id VARCHAR(50) DEFAULT NULL,
-            msg_type INTEGER DEFAULT NULL,
-            msg_talker VARCHAR(50) DEFAULT NULL,
-            msg_to VARCHAR(50) DEFAULT NULL,
-            msg_room VARCHAR(50) DEFAULT NULL,
-            msg_text TEXT DEFAULT NULL
-        );
-        """
-        await conn.execute(create_database_sql)
-        insert_table_sql = """
-        INSERT INTO ChatHistory (msg_date, msg_id, msg_type, msg_talker, msg_to, msg_room, msg_text) 
-        VALUES (?, ?, ?, ?, ?, ?, ?);
-        """
-        await conn.execute(insert_table_sql, message)
-        await conn.commit()
-        await conn.close()
+        async_engine = create_async_engine(self.chat_history_database)
+        async with async_engine.begin() as conn:
+            await conn.run_sync(Base.metadata.create_all)
+        async_session = sessionmaker(async_engine,
+                                     expire_on_commit=False,
+                                     class_=AsyncSession)
+        async with async_session() as session:
+            async with session.begin():
+                chathistroy = ChatHistory(
+                    msg_id=msg.message_id,
+                    msg_date=msg.date(),
+                    msg_type=msg.type(),
+                    msg_room=str(msg.room()) if msg.payload.room_id else None,
+                    msg_talker=str(
+                        msg.talker()) if msg.payload.from_id else None,
+                    msg_receiver=str(msg.to()) if msg.payload.to_id else None,
+                    msg_text=msg.text()
+                )
+                session.add(chathistroy)
+            await session.commit()
+        await async_engine.dispose()
```

Comment:
It will init the db every time when there is a new message ? The initialize related work shoule be done in [`init_plugin`](https://github.com/wechaty/python-wechaty/blob/master/src/wechaty/plugin.py#L102) method.

How do you think about it ?

Reply:
Right, move the db initialization to `init__plugin`.